server/gameconnection.py

```diff
@@ -294,44 +298,76 @@ async def handle_game_result(self, army, result):
         except (KeyError, ValueError):  # pragma: no cover
             self._logger.warning("Invalid result for %s reported: %s", army, result)
 
-    async def handle_operation_complete(self, army, secondary, delta):
-        if not int(army) == 1:
+    async def handle_operation_complete(
+        self, primary: Any, secondary: Any, delta: str
+    ):
+        """
+        :param primary: are primary mission objectives complete?
+        :param secondary: are secondary mission objectives complete?
+        :param delta: the time at which the operation occurred
+        :return: None
+        """
+        primary = FA.ENABLED == primary
+        secondary = FA.ENABLED == secondary
+
+        if not primary:
+            return
+
+        if not isinstance(self.game, CoopGame):
+            self._logger.warning(
+                "OperationComplete called for non-coop game: %s", self.game.id
+            )
             return
 
         if self.game.validity != ValidityState.COOP_NOT_RANKED:
+            self._logger.warning(
```

Comment:
I wouldn't log at all here. It's perfectly normal for people to play unranked games. For coop this could be as simple as setting the difficulty to normal or easy. I consider warnings to be more like configuration errors, or like above, code states that shouldn't be possible but aren't fatal.